pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go

```diff
@@ -23,6 +23,68 @@ import (
 // For name translation only please use metricNamesMapper instead
 type metricTransformerFunc = func(sender.Sender, string, ksmstore.DDMetric, string, []string, time.Time)
 
+var renamedResource = map[string]string{
+	"nvidia_com_gpu":     "gpu",
+	"amd_com_gpu":        "gpu",
+	"gpu_intel_com_i915": "gpu",
+	// Note: the following does not work out of the box, because it is filtered out of KSM metrics by default.
+	//       and needs to be grabbed some other way. At time of commit, this is via customresources/node.go.
+	//       More info: https://github.com/kubernetes/kube-state-metrics/issues/2027
+	"kubernetes_io_network_bandwidth": "network_bandwidth",
+}
+
+var nodeAllowedResources = map[string]struct{}{
+	"cpu":               {},
+	"memory":            {},
+	"pods":              {},
+	"ephemeral_storage": {},
+	"network_bandwidth": {},
+	"gpu":               {},
+}
+
+var containerAllowedResources = map[string]struct{}{
+	"cpu":               {},
+	"memory":            {},
+	"network_bandwidth": {},
+	"gpu":               {},
+}
+
+func renameResource(resource string) string {
+	if strings.HasPrefix(resource, "nvidia_com_mig") {
+		return "gpu"
+	}
+
+	if renamed, found := renamedResource[resource]; found {
+		return renamed
+	}
+
+	return resource
+}
+
+func resourceExtraTag(resource string) []string {
+	if strings.HasPrefix(resource, "nvidia_com_mig") {
+		splitStr := strings.Split(resource, "mig_")
+		if len(splitStr) == 2 {
+			return []string{fmt.Sprintf("mig_profile:%s", strings.Replace(splitStr[1], "_", "-", 2))}
+		}
+	}
```

Comment:
The use of `strings.Split(…)` looks a little bit overkill as, thanks to the `if strings.HasPrefix(…)`, we already know what the first token will be.
Basically, if we matched the `nvidia_com_mig` prefix, what we need is just what is after.
Also, is it on purpose that we replace only the first two `_` in `-` ?
I looked at the tests and couldn’t find a case where we would be interested in keeping a potential third `_` as is.

```suggestion
	if suffix, found := strings.CutPrefix(resource, "nvidia_com_mig_"); found {
		return []string{"mig_profile:" + strings.ReplaceAll(suffix, "_", "-")}
	}
```

Reply:
Good catch. `CutPrefix` is indeed what I wanted to build. I replaced only the first two "_" because normally there shouldn't be a third `_` but indeed we can replace all of them.